src/core/reference/include/openvino/reference/col2im.hpp

```diff
@@ -48,10 +48,10 @@ void col2im(const T* data,
             const auto channel_idx = column / kernel_product;
 
             for (int64_t column_height_idx = 0; column_height_idx < original_height; ++column_height_idx) {
-                const int64_t image_height_idx = get_image_dimension_index(column_height_idx, height_offset, 0);
+                const T_idx image_height_idx = get_image_dimension_index(column_height_idx, height_offset, 0);
```

Comment:
In some tests I was getting a compiler warning about comparison of variables of different signedness. Modifying the type to `T_idx` fixed the issue.

Reply:
The `get_image_dimension_index` should return i64 if T_idx is i32 then there is a risk the value will be lost as i64 will be casted to i32. It will just mask the compare issue.

Reply:
Fixed, the type `int64_t` is back.